packages/loader/container-loader/src/container.ts

```diff
@@ -1993,10 +1995,19 @@ export class Container
 				durationFromDisconnected =
 					time - this.connectionTransitionTimes[ConnectionState.Disconnected];
 				durationFromDisconnected = TelemetryLogger.formatTick(durationFromDisconnected);
+				// On first connection, also note how much gap was there between last processed seq number which would
+				// mostly be JoinOp and the snapshot seq number.
+				if (this._snapshotSeqNumber !== undefined && this.firstConnection) {
+					opsBehind = this.deltaManager.lastSequenceNumber - this._snapshotSeqNumber;
```

Comment:
Not sure why you are using this.deltaManager.lastSequenceNumber over this.deltaManager.lastKnownSeqNumber. I think later is correct in all cases

Reply:
BTW, I'd use the same formular (checkpointSequenceNumber - this.deltaManager.lastSequenceNumber) in all cases - it looks correct to me in a sense of intention - we are recording how far we are behind at the moment of connection. The only thing that is wrong in main - lastSequenceNumber = 0 when connection established before we loaded snapshot.

I'd actually do something like that:

```ts
if (this.deltaManager.hasCheckpointSequenceNumber && this._lifecycleState == "loaded") {
    opsBehind = checkpointSequenceNumber - this.deltaManager.lastSequenceNumber;
}
```

I'd probably not record anything on connected event. We are never behind for "write" connections. For "read" connections we might find it useful to record where we are, but same will be captured for CatchingUp event, and in future we plan to switch "read" to behave like "write", so not much value for this code to record anything.